app/src/main/java/emu/skyline/preference/IntegerListPreference.kt

```diff
@@ -0,0 +1,329 @@
+
+    private var value : Int? = null
+        set(value) {
+            // Always persist/notify the first time.
+            val changed = field != value
+            if (changed || !isValueSet) {
+                field = value
+                isValueSet = true
+                value?.let { persistInt(it) }
+                if (changed) {
+                    notifyChanged()
+                }
+            }
+        }
+
+    private var isValueSet = false
+    val refreshRequired : Boolean
+
+    init {
+        val res : Resources = context.resources
+
+        val a = context.obtainStyledAttributes(
+            attrs, sR.styleable.IntegerListPreference, defStyleAttr, defStyleRes
+        )
+
+        entries = TypedArrayUtils.getTextArray(
+            a, sR.styleable.IntegerListPreference_entries,
+            sR.styleable.IntegerListPreference_android_entries
+        )
+
+        val entryValuesId = TypedArrayUtils.getResourceId(
+            a, sR.styleable.IntegerListPreference_android_entryValues,
+            sR.styleable.IntegerListPreference_android_entryValues, 0
+        )
+        entryValues = if (entryValuesId != 0) res.getIntArray(entryValuesId) else null
+
+        if (TypedArrayUtils.getBoolean(
+                a, sR.styleable.IntegerListPreference_useSimpleSummaryProvider,
+                sR.styleable.IntegerListPreference_useSimpleSummaryProvider, false
+            )
+        ) {
+            summaryProvider = SimpleSummaryProvider.instance
+        }
+
+        refreshRequired = TypedArrayUtils.getBoolean(
+            a, sR.styleable.IntegerListPreference_refreshRequired,
+            sR.styleable.IntegerListPreference_refreshRequired, false
+        )
+
+        a.recycle()
+    }
+
+    /**
+     * @param entriesResId The entries array as a resource
+     */
+    fun setEntries(@ArrayRes entriesResId : Int) {
+        entries = context.resources.getTextArray(entriesResId)
+    }
+
+    /**
+     * @param entryValuesResId The entry values array as a resource
+     */
+    fun setEntryValues(@ArrayRes entryValuesResId : Int) {
+        entryValues = context.resources.getIntArray(entryValuesResId)
+    }
+
+    /**
+     * @return The entry corresponding to the current value, or `null`
+     */
+    fun getEntry() : CharSequence? {
+        return entries?.getOrNull(getValueIndex())
+    }
+
+    /**
+     * @param value The value whose index should be returned
+     * @return The index of the value, or -1 if not found
+     */
+    private fun findIndexOfValue(value : Int?) : Int {
+        entryValues?.let {
+            if (value != null) {
+                for (i in it.indices.reversed()) {
+                    if (it[i] == value) {
+                        return i
+                    }
+                }
+            }
+        }
+        return value ?: -1
+    }
+
+    /**
+     * Sets the value to the given index from the entry values.
+     * @param index The index of the value to set
```

Comment:
A fairly redundant comment in my opinion, do consider if someone reading this might spend more time reading the comment than figuring out what it means based on the naming